src/Update.hs

```diff
@@ -221,19 +220,22 @@ updatePackage log updateEnv mergeBaseOutpathsContext =
     let msgs = catMaybes [msg1, msg2]
     ----------------------------------------------------------------------------
     --
-    -- Compute the diff, look at rebuilds, and publish the package
+    -- Compute the diff and get updated values
     diffAfterRewrites <- Git.diff
     lift . log $ "Diff after rewrites::\n" <> diffAfterRewrites
     updatedDerivationContents <- liftIO $ T.readFile derivationFile
+    newSrcUrl <- Nix.getSrcUrl attrPath
+    newHash <- Nix.getHashFromBuild attrPath
```

Comment:
I'm a little wary of moving this so far from the lines that set the hash to all zeros because this only works if the hash is currently wrong.

Reply:
Ooof, didn't realize that ... let me take a closer look and actually try to test this.

Reply:
Ok, I've made two changes and given it a dry-run test on a package or two successfully.

1. Updated the `version` rewriter to still do this assert, and throw if it fails.  This covers the case where it enters into the `version` rewriter but fails to make a material change.

2. Updated the main to just call `Nix.getHash`, which always works to get the hash, and assert different after rewrites. This catches the case where `version` skipped it but no other rewriter made a material change.

Check (1) is probably redundant with (2) around, but it's probably best to be conservative.  As it stands, this gives the refactoring to let us add Go/Rust rewriters that can bump multi-fixed-output derivation files.